framework/src/constraints/AutomaticMortarGeneration.C

```diff
@@ -331,6 +343,11 @@ AutomaticMortarGeneration::buildMortarSegmentMesh()
     if (current_mortar_segment == nullptr)
       mooseError("Unable to find appropriate mortar segment during linear search!");
 
+    // // If node lands on endpoint of segment, don't split
+    // info = &msm_elem_to_info.at(current_mortar_segment);
+    // if (info->xi1_a == xi1 || xi1 == info->xi1_b)
+    //   continue;
+
```

Comment:
This was part of the fix to that one guy's 2D mortar problem with the funky geometry. For that fix we also change the signs a couple lines above in `(info->xi1_a < xi1 && xi1 < info->xi1_b)` to `<=`. This got taken out while I was trying to diagnose those ironing issues in 2D (where I was using the invalid quadratures), it could be put back in if you with but it was a very funky case that probably doesn't need to be handled.

Reply:
Hmm in that case, let's either remove the lines and explain what they were for in the commit message, or add another comment about what this might be used for (e.g. a weird geometry with duplicate notes ... that's what that person had right?)

Reply:
I put it back in, is the original comment not descriptive enough (that the element won't be split if the point lands on an endpoint)? Yeah it was a problem of two nodes projecting onto the exact same point.